configure

```diff
@@ -0,0 +1,459 @@
+#!/usr/bin/env dmd -run configure
+module configure;
+
+import std.algorithm;
+import std.array;
+import std.conv;
+import std.exception;
+import std.file;
+import std.format;
+import std.getopt;
+import std.path;
+import std.process;
+import std.range;
+import std.stdio : println = writeln;
+import std.string;
+import std.uni;
+import std.traits;
+
+version (Posix)
+{
+    version (OSX) {}
+    else version (linux) {}
+    else version (FreeBSD) {}
+    else
+        static assert("The current platform is not supported");
+}
+else
+    static assert("This script should only be run on Posix platforms");
+
+struct Options
+{
+    bool verbose = true;
+    bool help = false;
+
+    string llvmConfigPath;
+    string ncursesLibPath;
+
+    bool dynamicClang;
+}
+
+struct DefaultConfig
+{
+static:
+
+    version (OSX)
+    {
+        enum cppLib = "c++";
+        enum clangLib = "libclang.dylib";
+
+        immutable llvmConfigPaths = [
+            "/opt/local/bin/llvm-config-mp-3.9", // MacPorts
+            "/usr/local/opt/llvm39/bin/llvm-config", // Homebrew
+            "/opt/local/bin/llvm-config-mp-3.8", // MacPorts
+            "/usr/local/opt/llvm38/bin/llvm-config", // Homebrew
+            "llvm-config" // Fallback to whatever is in PATH
+        ];
+
+        immutable systemLibPahts = [
+            "/opt/local/lib", //MacPorts
+            "/usr/local/lib", // Homebrew
+            "/usr/lib", // System
+            "lib" // Local
+        ];
+    }
+
+    else version (linux)
+    {
+        enum cppLib = "stdc++";
+        enum clangLib = "libclang.so";
+
+        immutable llvmLibPaths = [
+            "/usr/lib/llvm-3.9/lib",
```

Comment:
Yes, these are the default paths it will look for. There's also a flag to explicitly specify which LLVM to use. I guess the script could iterate `/usr/lib/` and look for `llvm-x.x`. Seems a bit more complex though.